src/RepositoryProxy.php

```diff
@@ -8,13 +8,15 @@
 use PHPUnit\Framework\Assert;
 
 /**
- * @mixin EntityRepository
+ * todo: this mixin already existed, but with another syntax. Will this break something?
+ * @mixin EntityRepository<TProxiedObject>
```

Comment:
Psalm doesn't supported a prefixed `psalm-mixin` annotation. I wonder if we break something that was working before by using this syntax?

Reply:
I only added as a helper (for PhpStorm) because most people's repository's would be an instance of `EntityRepository` but this may not be the case. For instance, in my apps, my repositories are instances of `ObjectRepository` but not `EntityRepository`.

Does PhpStorm still autocomplete entity repository with this change? That would be my only concern.

Reply:
Just tested it by trying to autocomplete `UserFactory::repository()->createQueryBuilder()` (and removing the `UserRepository|RepositoryProxy repository()` PHPdoc on the `UserFactory`). This works fine, so I think PHPstorm ignores the `<...>` part.